gala/potential/potential/builtin/builtin_potentials.h

```diff
@@ -1,98 +1,100 @@
+#include <stddef.h>
+
 extern double nan_density(double t, double *pars, double *q, int n_dim, void *state);
 extern double nan_value(double t, double *pars, double *q, int n_dim, void *state);
-extern void nan_gradient(double t, double *pars, double *q, int n_dim, double *grad, void *state);
+extern void nan_gradient(size_t N, double t, double *pars, double *q, int n_dim, double *grad, void *state);
```

Comment:
I must have missed this from looking above, but why do the gradient functions need to take in `N`? And for consistency, would it make more sense to keep `t` as the first argument and take `N` after `pars` or something?